pinecone/management_client.go

```diff
@@ -0,0 +1,488 @@
+package pinecone
+
+import (
+	"context"
+	"fmt"
+	"github.com/deepmap/oapi-codegen/v2/pkg/securityprovider"
+	"github.com/google/uuid"
+	"github.com/pinecone-io/go-pinecone/internal/gen/management"
+	"net/http"
+)
+
+// ManagementClient provides a high-level interface for interacting with the
+// Pinecone management plane API. It encapsulates the necessary authentication,
+// request creation, and response handling for the API's operations.
+//
+// The ManagementClient is designed to simplify the management of projects and
+// their API keys by abstracting away the direct handling of HTTP requests and
+// responses. It leverages a generated low-level client (`restClient`) for
+// communication with the API, ensuring that requests are properly authenticated
+// and formatted according to the API's specification.
+//
+// Fields:
+//   - apiKey: The API key used for authenticating requests to the management API.
+//     This key should have the necessary permissions for the operations you intend to perform.
+//   - restClient: An instance of the generated low-level client that actually performs
+//     HTTP requests to the management API. This field is internal and managed
+//     by the ManagementClient.
+//
+// To use ManagementClient, first instantiate it using the NewManagementClient function,
+// providing it with the necessary configuration. Once instantiated, you can call its
+// methods to perform actions such as listing, creating, getting, and deleting projects
+// and project API keys.
+//
+// Example:
+//
+//	clientParams := NewManagementClientParams{ApiKey: "your_api_key_here"}
+//	managementClient, err := NewManagementClient(clientParams)
+//	if err != nil {
+//	    log.Fatalf("Failed to create management client: %v", err)
+//	}
+//	// Now you can use managementClient to interact with the API
+//
+// Note that ManagementClient methods are designed to be safe for concurrent use by multiple
+// goroutines, assuming that its configuration (e.g., the API key) is not modified after
```

Comment:
Is there a way to make the API key protected to ensure this? Something like Java's `private`?